dhis-2/dhis-web-api/src/main/java/org/hisp/dhis/webapi/controller/AccountController.java

```diff
@@ -481,26 +461,22 @@ public WebMessage createAccount(
     }
 
     @PostMapping( "/password" )
-    public void updatePassword(
+    public ResponseEntity<Map<String, String>> updatePassword(
         @RequestParam String oldPassword,
         @RequestParam String password,
-        HttpServletRequest request,
-        HttpServletResponse response )
-        throws IOException
+        User currentUser,
+        HttpServletRequest request )
     {
-        String username = (String) request.getSession().getAttribute( "username" );
-        UserCredentials credentials = userService.getUserCredentialsByUsername( username );
+        String username = currentUser.getUsername();
```

Comment:
The `request.getSession().getAttribute( "username" )` would not be set properly in a controller test so I used injection of the `User currentUser` instead assuming this is the same in a production system as this also works in tests. In addition this also limits the need to use `HttpServletRequest` directly. There is still one other use of `request` but we might be able to clean that up at some later point as well.